ddtrace/contrib/internal/langchain/patch.py

```diff
@@ -975,6 +977,138 @@ def traced_similarity_search(langchain, pin, func, instance, args, kwargs):
+        on_span_started=_on_span_started,
+        on_span_finished=_on_span_finished,
+    )
+
+
+@with_traced_module
+def traced_chat_stream(langchain, pin, func, instance, args, kwargs):
+    integration: LangChainIntegration = langchain._datadog_integration
+    llm_provider = instance._llm_type
+
+    def _on_span_started(span: Span):
+        if not integration.is_pc_sampled_span(span):
+            return
+        chat_messages = get_argument_value(args, kwargs, 0, "input")
+        tag_general_message_input(span, chat_messages, integration, langchain_core)
+
+        for param, val in getattr(instance, "_identifying_params", {}).items():
+            if not isinstance(val, dict):
+                span.set_tag_str("langchain.request.%s.parameters.%s" % (llm_provider, param), str(val))
+                continue
+            for k, v in val.items():
+                span.set_tag_str("langchain.request.%s.parameters.%s.%s" % (llm_provider, param, k), str(v))
+
+    def _on_span_finished(span: Span, streamed_chunks):
+        if span.error or not integration.is_pc_sampled_span(span):
+            return
+        content = "".join([str(chunk.content) for chunk in streamed_chunks])
+        span.set_tag_str("langchain.response.content", integration.trunc(content))
+
+        usage = streamed_chunks and getattr(streamed_chunks[-1], "usage_metadata", None)
+        if not usage:
+            return
+        for k, v in usage.items():
+            span.set_tag_str("langchain.response.usage_metadata.%s" % k, str(v))
+
+    return shared_stream(
+        integration=integration,
+        pin=pin,
+        func=func,
+        instance=instance,
+        args=args,
+        kwargs=kwargs,
+        interface_type="chat_model",
+        on_span_started=_on_span_started,
+        on_span_finished=_on_span_finished,
+        api_key=_extract_api_key(instance),
+        provider=llm_provider,
+    )
+
+
+@with_traced_module
+def traced_llm_stream(langchain, pin, func, instance, args, kwargs):
+    integration: LangChainIntegration = langchain._datadog_integration
+    llm_provider = instance._llm_type
+
+    def _on_span_start(span: Span):
+        if not integration.is_pc_sampled_span(span):
+            return
+        inp = get_argument_value(args, kwargs, 0, "input")
+        tag_general_message_input(span, inp, integration, langchain_core)
+        for param, val in getattr(instance, "_identifying_params", {}).items():
+            if not isinstance(val, dict):
+                span.set_tag_str("langchain.request.%s.parameters.%s" % (llm_provider, param), str(val))
+                continue
+            for k, v in val.items():
+                span.set_tag_str("langchain.request.%s.parameters.%s.%s" % (llm_provider, param, k), str(v))
+
+    def _on_span_finished(span: Span, streamed_chunks):
+        if span.error or not integration.is_pc_sampled_span(span):
+            return
+        content = "".join([str(chunk) for chunk in streamed_chunks])
```

Comment:
Just for clarity: llm streams chunks are in the form of raw text?

Reply:
similar to the chat model case, yeah their `llm.stream` definition specifies it's an [iterator of strings](https://github.com/langchain-ai/langchain/blob/f9ef688b3a0e3f576a76435e242090a76666974b/libs/core/langchain_core/language_models/llms.py#L515-L522).